Stratis.Bitcoin.Features.RPC/RpcSettings.cs

```diff
@@ -58,58 +58,69 @@ public void Load(NodeSettings nodeSettings)
             var config = nodeSettings.ConfigReader;
 
             this.Server = config.GetOrDefault<bool>("server", false);
-            if (!this.Server)
-                return;
+            this.RPCPort = nodeSettings.Network.RPCPort;
+
+            if (this.Server)
+            {
+                this.RpcUser = config.GetOrDefault<string>("rpcuser", null);
+                this.RpcPassword = config.GetOrDefault<string>("rpcpassword", null);
+
+                var defaultPort = config.GetOrDefault<int>("rpcport", nodeSettings.Network.RPCPort);
+                this.RPCPort = defaultPort;
+
+                try
+                {
+                    this.AllowIp = config
+                        .GetAll("rpcallowip")
+                        .Select(p => IPAddress.Parse(p))
+                        .ToList();
+                }
+                catch (FormatException)
+                {
+                    throw new ConfigurationException("Invalid rpcallowip value");
+                }
+            }
+
+            this.callback?.Invoke(this);
 
-            this.RpcUser = config.GetOrDefault<string>("rpcuser", null);
-            this.RpcPassword = config.GetOrDefault<string>("rpcpassword", null);
             if (this.RpcPassword == null && this.RpcUser != null)
                 throw new ConfigurationException("rpcpassword should be provided");
             if (this.RpcUser == null && this.RpcPassword != null)
                 throw new ConfigurationException("rpcuser should be provided");
 
-            var defaultPort = config.GetOrDefault<int>("rpcport", nodeSettings.Network.RPCPort);
-            this.RPCPort = defaultPort;
-            try
-            {
-                this.Bind = config
-                    .GetAll("rpcbind")
-                    .Select(p => NodeSettings.ConvertToEndpoint(p, defaultPort))
-                    .ToList();
-            }
-            catch (FormatException)
-            {
-                throw new ConfigurationException("Invalid rpcbind value");
-            }
+            this.Server = true;
```

Comment:
why do we set the server to true here?

Reply:
Due to the preceding check the server variable would already have a value of true at that point - unless the rpc user and password were set via callback (in the typical "addRpc" scenario), in which case we would want the rpc server to be active.